agent-shell.el

```diff
@@ -2029,6 +2039,71 @@ When DEACTIVATE is non-nil, deactivate region/selection."
               "\n"
               "```"))))
 
+;;; Mode cycling
+
+(defun agent-shell--get-mode-name (mode-id available-modes)
+  "Get the name of the mode with MODE-ID from AVAILABLE-MODES.
+AVAILABLE-MODES is the list of mode objects from the ACP session/new response.
+Each mode has an 'id' and 'name' field. We look up the mode by ID to get its
+display name."
+  (when-let ((mode (cl-find mode-id available-modes
+                            :key (lambda (m) (map-elt m 'id))
+                            :test #'string=)))
+    (map-elt mode 'name)))
+
+(defun agent-shell--mode-line-format ()
+  "Return the mode-line format for displaying the current permission mode."
```

Comment:
Including a sample modeline in the docstring would be helpful for anyone reading the code.